homeassistant/components/number/__init__.py

```diff
@@ -357,7 +358,14 @@ def step(self) -> float:
     @property
     def mode(self) -> NumberMode:
         """Return the mode of the entity."""
-        return self._attr_mode
+        if hasattr(self, "_attr_mode"):
+            return self._attr_mode
+        if (
+            hasattr(self, "entity_description")
+            and self.entity_description.mode is not None
+        ):
```

Comment:
I believe this will create a problem as a mode of None is not supported and messes up the UI. Hence my return of mode AUTO on line 368

Reply:
And what does the `baf` integration have to do with the number entity? I also don't own such device

Reply:
> I believe this will create a problem as a mode of None is not supported and messes up the UI. Hence my return of mode AUTO on line 368

You need to change the default on `NumberEntityDescription` on line 136 and remove support for `None` there.

> And what does the `baf` integration have to do with the number entity? I also don't own such device

Your PR causes impacts the `baf` integration, and causes mypy + pylint failure.
You need to investigate this accordingly.

Reply:
I don't see how the `baf` integration is harmed by this change. I checked the `baf` code, and I don't see anything that relates to my change. Perhaps the `baf` integration was already failing before my change?